controller/api/api.go

```diff
@@ -612,8 +612,14 @@ func (a *Api) scaleContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := a.manager.ScaleContainer(containerId, numInstances); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	result := a.manager.ScaleContainer(containerId, numInstances)
+	if len(result.Errors) > 0 {
+		http.Error(w, string(json), http.StatusInternalServerError)
```

Comment:
This should be `result`.  `http.Error(w, result, http.StatusInternalServerError)` or whatever result is returning.